datastore/longhorn.go

```diff
@@ -557,6 +557,56 @@ func (s *DataStore) ListVolumes() (map[string]*longhorn.Volume, error) {
 	return itemMap, nil
 }
 
+func MarshalLabelToVolumeRecurringJob(labels map[string]string) map[string]*longhorn.VolumeRecurringJob {
+	groupPrefix := fmt.Sprintf(types.LonghornLabelRecurringJobKeyPrefixFmt, types.LonghornLabelRecurringJobGroup) + "/"
+	jobPrefix := fmt.Sprintf(types.LonghornLabelRecurringJobKeyPrefixFmt, types.LonghornLabelRecurringJob) + "/"
+	jobMapVolumeJob := make(map[string]*longhorn.VolumeRecurringJob)
+	for label := range labels {
+		if strings.HasPrefix(label, groupPrefix) {
+			jobName := strings.TrimPrefix(label, groupPrefix)
+			jobMapVolumeJob[jobName] = &longhorn.VolumeRecurringJob{
+				Name:    jobName,
+				IsGroup: true,
+			}
+			continue
+		} else if strings.HasPrefix(label, jobPrefix) {
+			jobName := strings.TrimPrefix(label, jobPrefix)
+			jobMapVolumeJob[jobName] = &longhorn.VolumeRecurringJob{
+				Name:    jobName,
+				IsGroup: false,
+			}
+		}
+	}
+	return jobMapVolumeJob
+}
+
+func (s *DataStore) DeleteVolumeRecurringJob(volumeName string, name string, isGroup bool) (v *longhorn.Volume, err error) {
+	defer func() {
+		err = errors.Wrapf(err, "failed to delete volume recurring jobs for %v", volumeName)
+	}()
+
+	v, err = s.GetVolume(volumeName)
```

Comment:
NIT: Why do we re-get `v *longhorn.Volume` rather than relying on the caller to input it? If you prefer to retain the current implementation, you can use `ListVolumesByXxxxxxRO` in the caller to avoid meaningless deep copies.

Reply:
Fixed.